pkg/controller/jobframework/reconciler.go

```diff
@@ -560,27 +560,37 @@ func (r *JobReconciler) constructWorkload(ctx context.Context, job GenericJob, o
 		)
 	}
 
-	priorityClassName, p, err := r.extractPriority(ctx, podSets, job)
+	priorityClassName, p, source, err := r.extractPriority(ctx, podSets, job)
 	if err != nil {
 		return nil, err
 	}
 
 	wl.Spec.PriorityClassName = priorityClassName
 	wl.Spec.Priority = &p
+	wl.Spec.PriorityClassSource = source
 
 	if err := ctrl.SetControllerReference(object, wl, r.client.Scheme()); err != nil {
 		return nil, err
 	}
 	return wl, nil
 }
 
-func (r *JobReconciler) extractPriority(ctx context.Context, podSets []kueue.PodSet, job GenericJob) (string, int32, error) {
+func (r *JobReconciler) extractPriority(ctx context.Context, podSets []kueue.PodSet, job GenericJob) (string, int32, string, error) {
+	if workloadPriorityClass, isWorkloadPriorityClass := WorkloadPriorityClassName(job); isWorkloadPriorityClass {
+		priorityClassName, p, err := utilpriority.GetPriorityFromWorkloadPriorityClass(ctx, r.client, workloadPriorityClass)
+		source := constants.WorkloadPriorityClassSource
+		return priorityClassName, p, source, err
+	}
+	source := constants.K8sPriorityClassSource
```

Comment:
If priorityClass is not workloadPriorityClass, can we label it as `k8sPriorityClassSource`?

Reply:
Do you mean that kueue will set `kueue.x-k8s.io/workloadpriorityclass` to `.spec.priorityClassSource` in the Workload if jobs don't have `kueue.x-k8s.io/priority-class` label?

Reply:
My question is about what should be set in the `PrirorityClassSource` field of `workload` if a job does not have a `priorityClassName`.
Within the `GetPriorityFromPriorityClass` function, if there is no default `priorityClass`, it returns a priority of 0. In this case, I am considering what value should be assigned to the `source` field of `workload`.

Reply:
Ah, I see. Thanks for the clarification.
Setting an empty source there would be better since there are situations not to set priority and we should keep backward compatibility.

Reply:
Thanks, fixed it